zap/src/main/java/org/parosproxy/paros/network/HttpMessage.java

```diff
@@ -1090,4 +1094,27 @@ public void setResponseFromTargetHost(final boolean responseFromTargetHost) {
     public String getType() {
         return MESSAGE_TYPE;
     }
+
+    public static void replaceCsrfToken(
+            HttpMessage msg,
+            AntiCsrfToken antiCsrfToken,
+            String tokenValue,
+            Logger log,
+            Encoder encoder,
+            ExtensionAntiCSRF extAntiCSRF) {
+        log.debug(
+                "regenerateAntiCsrfToken replacing "
```

Comment:
replace*

Reply:
This replace method may only currently be called from regenerate but there’s no way we can be sure that’ll always be the case. 
Also I think it’ll be more confusing than helpful for future debugging/ maintenance of the message isn’t aligned to the local most function name.

Reply:
So I should move the logging out of the method and have it instead being done right before the method invocation?

Reply:
Sure it could be moved into the regenerate method ahead of the call

Reply:
> replace*

Sorry I think I missed this, did you mean to replace the word "replacing" with "replace"?

Reply:
Probably doesn't apply anymore, this was when you'd implemented a replace method not a regenerate......

Reply:
Gotcha haha, thanks for the clarification :)